app/src/views/users/__tests__/login.spec.tsx

```diff
@@ -0,0 +1,70 @@
+import { render, fireEvent, screen, waitFor, cleanup } from "@testing-library/react";
+
+import Login from "../login";
+
+import * as auth from "../../../services/auth";
+
+import { withTestRouter } from "../../../utils/helpers/withTestRouter";
+
+jest.mock("../../../services/auth", () => ({
+  login: jest.fn(),
+}));
+
+describe("Login", () => {
+  afterEach(() => cleanup);
+  it("renders login form", async () => {
+    const { findByRole, findByLabelText } = render(<Login />);
+    const emailInputElement = await findByLabelText(/email/i);
+    const passwordInputElement = await findByLabelText(/password/i);
```

Comment:
What do you think about putting it directly into `expect`s? Those variables are used only once, so it would be shorter, but from the other side, this way is maybe more readable...

Reply:
Fair point, will do 😄